### PR TITLE
Fix golangci-lint issues and integrate the same on CI

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

This PR addresses some issues related to `golangci-lint` along with adding lint checks in CI since they were not being captured by our bot. Probably, since most bots tend to comment on the new changes, that might be a reason why these errors were not caught yet.

### CODE REVIEW

**Review by JLockerman — COMMENTED** [review]
Looks good to me for the changes to our code.
@cevian we need a policy decision on what to do with the code we copied from prom; I kinda want it to be as similar as possible to ease diffing when we want to upgrade.

**Comment by Harkishen-Singh** [comment]
> I wish there was a way to just golint our changes but I don't think there is?
I think that a regex should do that work.

**Comment by Harkishen-Singh** [comment]
> all the changes under pkg/promql need to be rolled back 

Done!

File: pkg/promql/functions_test.go [hunk 1]
```diff
@@ -67,6 +67,6 @@ func TestFunctionList(t *testing.T) {
 
 	for i := range parser.Functions {
 		_, ok := FunctionCalls[i]
-		testutil.Assert(t, ok, (fmt.Sprintf("function %s exists in parser package, but not in promql package", i)))
+		testutil.Assert(t, ok, fmt.Sprintf("function %s exists in parser package, but not in promql package", i))
```
Comment (cevian) [inline pkg/promql/functions_test.go:70]: This error exists in upstream and should also be rolled back

**Review by cevian — APPROVED** [review]
One small comment to address otherwise good to merge

File: pkg/promql/functions_test.go (hunk 1 above)
Comment (Harkishen-Singh) [reply pkg/promql/functions_test.go:70]: oh, I thought its just us, so I let it be. Anyways, rolling back that too.

Comment (Harkishen-Singh) [reply pkg/promql/functions_test.go]: Done!

**Review by JLockerman — APPROVED** [review]

**Review by cevian — APPROVED** [review]
Please clean up the commit history before merging

**Comment by Harkishen-Singh** [comment]
@cevian I hope this looks good.

**Review by cevian — APPROVED** [review]

**Pull request merged**